src/services/oauth2.js

```diff
@@ -36,22 +38,24 @@ export class OAuth2Service {
   updateUser(user, data) {
     const idField = this.options.user.idField;
     const id = user[idField];
+    const userService = typeof this._userService === 'string' ? this.app.service(this._userService) : this._userService;
```

Comment:
Hmm. I don't remember that being an issue but you could be right that order might matter. In any case this is totally good for now. I might double check that order matters when it is in the `setup` method but it's pretty low priority right now. 👍